gulp-tasks/test.js

```diff
@@ -1,47 +1,69 @@
 const gulp = require('gulp');
 const oneLine = require('common-tags').oneLine;
+const glob = require('glob');
+const fse = require('fs-extra');
+const path = require('path');
 
 const spawn = require('./utils/spawn-promise-wrapper');
 const getNpmCmd = require('./utils/get-npm-cmd');
 const logHelper = require('../infra/utils/log-helper');
 
-const runNodeTestsWithEnv = (nodeEnv) => {
-  return () => {
-    logHelper.log(oneLine`
-      Running Node Tests.
-    `);
-    const options = [];
-    if (global.cliOptions.grep) {
-      options.push('--grep', global.cliOptions.grep);
-    }
-    const originalNodeEnv = process.env.NODE_ENV;
-
-    process.env.NODE_ENV = nodeEnv;
-    return spawn(getNpmCmd(), ['run', 'test', '--',
-      `./test/${global.packageOrStar}/node/**/*.{js,mjs}`,
+const runNodeTestSuite = async (testPath, nodeEnv) => {
+  logHelper.log(oneLine`
+  Running Node test on ${logHelper.highlight(testPath)}
+  with NODE_ENV '${nodeEnv}'
+`);
+
+  const options = [];
+  if (global.cliOptions.grep) {
+    options.push('--grep', global.cliOptions.grep);
+  }
+  const originalNodeEnv = process.env.NODE_ENV;
+
+  process.env.NODE_ENV = nodeEnv;
+  try {
+    await spawn(getNpmCmd(), ['run', 'test', '--',
+      `${testPath}/node/**/*.{js,mjs}`,
       ...options,
-    ])
-    .catch((err) => {
-      logHelper.error(err);
-      throw new Error(`[Workbox Error Msg] 'gulp test' discovered errors.`);
-    })
-    .then(() => {
-      process.env.NODE_ENV = originalNodeEnv;
-    });
-  };
+    ]);
+    process.env.NODE_ENV = originalNodeEnv;
+  } catch (err) {
+    process.env.NODE_ENV = originalNodeEnv;
+
+    logHelper.error(err);
+    throw new Error(`[Workbox Error Msg] 'gulp test' discovered errors.`);
+  }
+};
+
+const runNodeTestsWithEnv = (nodeEnv) => {
+  const packagesToTest = glob.sync(`test/${global.packageOrStar}`);
+  return packagesToTest.reduce(async (promiseChain, packageToTest) => {
```

Comment:
One of the nicer parts of `async`/`await` [you don't have to use `reduce()` chains anymore](https://developers.google.com/web/fundamentals/primers/async-functions#example_outputting_fetches_in_order):

```js
const runNodeTestsWithEnv = async (nodeEnv) => {
  const packagesToTest = glob.sync(`test/${global.packageOrStar}`);
  for (const packageToTest of packagesToTest) {
    await runNodeTestSuite(packageToTest, nodeEnv);
  }
};
```